framework/src/main/java/org/checkerframework/common/accumulation/AccumulationAnnotatedTypeFactory.java

```diff
@@ -135,6 +135,14 @@ public AccumulationTreeAnnotator(AccumulationAnnotatedTypeFactory factory) {
             super(factory);
         }
 
+        /**
+         * If a method returns its receiver, and the receiver has an accumulation type, then the
```

Comment:
Wait, isn't the point that this annotator takes the receiver type and "accumulates in" the current method invocation?

Reply:
Only the Object Construction Checker does that. This version is simpler - all it does it make the return type match the receiver type if the method returns `this`. The `TestAccumulationChecker` handles actually accumulating method calls differently than the OCC, also - note that it does the actual accumulation in the `Transfer` class.

Reply:
I get it now.  Since the transfer function computes the complete accumulated type for the most deeply-nested receiver, all the type factory has to do is to copy that type to the return value for each call and it will get bubbled up.